src/Radicle/Internal/Core.hs

```diff
@@ -440,6 +455,13 @@ instance GhcExts.IsList (PrimFns m) where
     fromList xs = PrimFns . Map.fromList $ [ (i, Doc.Docd d x)| (i, d, x) <- xs ]
     toList e = [ (i, d, x) | (i, Doc.Docd d x) <- GhcExts.toList . getPrimFns $ e]
 
+data PureState = PureState
+  { env  :: Env Value
+  , refs :: IntMap Value
+  } deriving (Eq, Ord, Read, Show, Generic)
```

Comment:
Can we find a better name than `PureState`? Maybe just `State` or `EvaluatorState`? What about prefixing the fields to avoid name clashes?